src/Servers/Kestrel/Core/src/Internal/HttpsConnectionMiddleware.cs

```diff
@@ -281,5 +260,19 @@ private static X509Certificate2 ConvertToX509Certificate2(X509Certificate certif
 
             return new X509Certificate2(certificate);
         }
+
+        private class SslDuplexPipe : DuplexPipeStreamAdapter<SslStream>
```

Comment:
@sebastienros I'd like to run a performance test with this change. It could affect SSL performance.